File: 2025/03/ec.go
package main

import (
	"fmt"
	"sort"

	. "github.com/beanz/everybodycodes/lib-go"
)

func parts(i1, i2, i3 []byte) (int, int, int) {
	num := []int{}
	for i := 0; i < len(i1); {
		j, n := ChompUInt[int](i1, i)
		i = j + 1
		num = append(num, n)
	}
	sort.Ints(num)
	p1 := num[0]
	p := num[0]
	for i := 1; i < len(num); i++ {
		if p == num[i] {
			continue
		}
		p = num[i]
		p1 += p
	}

	num = num[:0]
	for i := 0; i < len(i2); {
		j, n := ChompUInt[int](i2, i)
		i = j + 1
		num = append(num, n)
	}
	sort.Ints(num)
	p2 := num[0]
	p = num[0]
	c := 1
	for i := 1; i < len(num); i++ {
		if p == num[i] {
			continue
		}
		c++
		p = num[i]
		p2 += p
		if c == 20 {
			break
		}
	}

	num = num[:0]
	for i := 0; i < len(i3); {
		j, n := ChompUInt[int](i3, i)
		i = j + 1
		num = append(num, n)
	}
	sort.Ints(num)
	p3 := 0
	for i := 0; i < len(num); {
		n := num[i]
		c := 0
		for ; i < len(num); i++ {
			if num[i] == n {
				c++
			} else {
				break
			}
		}
		if p3 < c {
			p3 = c
		}
	}
	return p1, p2, p3
}

func main() {
	p1, p2, p3 := parts(Input(1, ""), Input(2, ""), Input(3, ""))
	fmt.Printf("Part 1: %v\nPart 2: %v\nPart 3: %v\n", p1, p2, p3)
}

File: 2025/03/ec.pl
#!/usr/bin/env perl
use strict;
use warnings;
use v5.30;
use lib "../../lib-perl";
no warnings 'portable';
use Helpers qw/:all/;

sub parse {
  my ($c) = @_;
  chomp($c);
  return split /,/, $c;
}

my @c = parse(input(1));
@c = uniq sort {$b<=>$a} @c;
print "Part 1: ", (sum @c), "\n";

@c = parse(input(2));
@c = uniq sort {$a<=>$b} @c;
print "Part 2: ", (sum @c[0..19]), "\n";

@c = parse(input(3));
my %m;
for (@c) { $m{$_}++ }
my $c = max values %m;
print "Part 3: ", $c, "\n";

